### PR TITLE
add branch name into commit textarea placeholder

### LINKED ISSUE #6114: [scm|git] update commit textarea placeholder to include branch name
**Description**

Update the git commit textarea placeholder to include the branch name.

Currently the following is displayed:

Screen Shot 2019-09-05 at 8 54 13 AM

While the placeholder can be updated to display the following:

Screen Shot 2019-09-05 at 9 00 06 AM

**Additional Info**

The feature was recently added in VSCode:
https://code.visualstudio.com/updates/v1_38#_source-control

---

**Bonus** 🥇 

Display ⌘ instead of Cmd for macOS

### PR DESCRIPTION
**Pull request by xieyuanhuata**
Signed-off-by: xieyuanhu <xieyuanhuata@163.com>

#### What it does
fix #6114  update commit textarea placeholder to include branch name

#### How to test
1、pull this commit and start the theia;
2、open 'Source control :Git';

#### Review checklist

- [ ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
@vince-fugnitto Should it be updated when a branch is changed?

**Comment by xieyuanhuata** [comment]
@akosyakov 
ok

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto Should it be updated when a branch is changed?

Yes, I believe so. I'd make sense to adjust the placeholder whenever a branch is switched.
Example: _VSCode_

1. Initially my branch is `master`:

<div align='center'>

Screen Shot 2019-09-11 at 8 50 48 AM

</div>

2. Switch to `GH-4162` (the placeholder is updated)

<div align='center'>

Screen Shot 2019-09-11 at 8 51 06 AM

</div>

**Comment by vince-fugnitto** [comment]
@xieyuanhuata I believe you can update the logic here to build the entire message:

https://github.com/theia-ide/theia/blob/98f12a9fcc3a43c2b5d224e186f871e66a1e11b7/packages/scm/src/browser/scm-widget.tsx#L185-L196

This will allow you to easily call `update` as well whenever a repository is changed.
I'm not sure it's possible to do it at the moment where you `registerScmProviders`.

**Comment by xieyuanhuata** [comment]
@vince-fugnitto 
Yes, you are right.I will modify the code.

File: packages/git/src/browser/git-repository-provider.ts [hunk 1]
```diff
@@ -185,11 +186,12 @@ export class GitRepositoryProvider {
...
     }
 
-    protected registerScmProvider(repository: Repository): void {
+    protected async registerScmProvider(repository: Repository): Promise<void> {
         const provider = this.scmProviderFactory({ repository });
+        const branch = await this.git.branch(repository, { type: 'current' });
         this.scmService.registerScmProvider(provider, {
             input: {
-                placeholder: 'Message (press {0} to commit)',
+                placeholder: `Message (${OS.type() === OS.Type.OSX ? '⌘' : 'press '}{0} to commit ${branch ? 'on \'' + branch.name + '\'' : ''})`,
```
Comment (akosyakov) [inline packages/git/src/browser/git-repository-provider.ts]: one should use [Keybindings.acceleratorFor](https://github.com/eclipse-theia/theia/blob/98f12a9fcc3a43c2b5d224e186f871e66a1e11b7/packages/core/src/browser/keybinding.ts#L307)  to create a human readable representation

**Comment by svenefftinge** [comment]
@xieyuanhuata would be great to have this change! Are you going to update the PR?

**Comment by westbury** [comment]
I have pushed a second commit that resolves the outstanding issues on this PR.

- The Keybindings.acceleratorFor was already being used, substituted into {0}.  I believe the '⌘Cmd' problem must be a pre-existing bug not related to this issue, but I don't have a Mac.
- We are limited in the way we listen to changes to the branch because we must stick to the VS Code API.  That API abstracts out any concept of branches, but we get the placeholder text from the builtin.  To get the placeholder text to work with the builtin, the changes in scm-commit-widget are required.  The changes to the two git files bring Theia's git package into alignment with the VS Code API.

**Comment by vince-fugnitto** [comment]
@westbury would you mind resolving the conflict and squashing the commits? (I can do so as well if necessary)
I will perform a review :)

**Comment by westbury** [comment]
@vince-fugnitto I can't squash the two commits because I am adding to the original pull-request so two authors.  The two commits should be merged separately.  I have however rebased.  I was not sure I would be able to force-push to someone else's fork, but it seemed to allow me.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto I can't squash the two commits because I am adding to the original pull-request so two authors. The two commits should be merged separately. I have however rebased. I was not sure I would be able to force-push to someone else's fork, but it seemed to allow me.

I was thinking one commit with multiple authors or co-authors since logically the pull-request should not be broken down and if we go back in history one of the commits would not be complete. The pull-request as is still has conflicts.

It would also be possible to open a new pull-request and add @xieyuanhuata as a co-author (but the pull-request comes from your branch).

**Comment by westbury** [comment]
> I was thinking one commit with multiple authors or co-authors

I don't think the Eclipse IP checker can process such merged commits.  See for example https://github.com/eclipse-theia/theia/pull/4279 which was merged as four commits due to multiple authors, and see comment https://github.com/eclipse-theia/theia/pull/4279#issuecomment-492505558 by @akosyakov .  The e-mail in the commit must be match both the sign-off and the Foundation's records, and there can only be one author e-mail per commit.

> The pull-request as is still has conflicts.

Sorry, pushed to wrong remote.

**Comment by vince-fugnitto** [comment]
> I don't think the Eclipse IP checker can process such merged commits. See for example #4279 which was merged as four commits due to multiple authors, and see comment #4279 (comment) by @akosyakov . The e-mail in the commit must be match both the sign-off and the Foundation's records, and there can only be one author e-mail per commit.

I see! In our case if one helped out with a pull-request we generally do the following a commit: https://github.com/eclipse-theia/theia/pull/8342/commits. Of course only the author will have the credit for GitHub.

I'll leave it up to your discretion however :) I'll perform a review shortly.

**Comment by westbury** [comment]
@vince-fugnitto You're right.  It looks like Eclipse do allow Co-authored-by.  I've merged the commits.

**Review by vince-fugnitto — APPROVED** [review]
@westbury the changes work very well, thank you for taking care of the issue 👍 
I verified the feature using both `@theia/git` and the builtins, with single and multi-repository workspaces.

**Pull request merged**